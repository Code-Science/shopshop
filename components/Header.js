import React, { useState } from 'react';
import Link from 'next/link';
import ShoppingCartOutlinedIcon from '@material-ui/icons/ShoppingCartOutlined';
import MenuIcon from '@material-ui/icons/Menu';
import { signIn, signOut, useSession } from 'next-auth/client';
import Nav from './Nav';
import Cart from './Cart';
import Backdrop from './Backdrop';
import { useStateValue } from '../store/StateProvider';
import styles from '../styles/components/Header.module.css';

function Header() {
  const [session] = useSession();
  const [openNav, setNav] = useState(false);
  const [showCart, setCart] = useState(false);
  const [{ basket }] = useStateValue();

  const openOrCloseNav = () => {
    setNav((prevState) => !prevState);
  };

  const showCartHandler = () => {
    setCart(true);
  };

  const closeCartHandler = () => {
    setCart(false);
  };

  const totalBasketItems = () => {
    return basket.reduce((sum, curr) => {
      return sum + curr.quantity;
    }, 0);
  };

  return (
    <div className={styles.header}>
      <div className={styles.auth}>
        {!session && (
          <button type="button" className={styles.items} onClick={signIn}>
            Sign In
          </button>
        )}
        {session && (
          <>
            <p className={styles.items}>{session.user.name}</p>
            <button type="button" className={styles.items} onClick={signOut}>
              SIGN OUT
            </button>
            <Link href="/accounts/orders">
              <a className={styles.items}>orders</a>
            </Link>
          </>
        )}
      </div>
      <div className={styles.mainHead}>
        <div className={styles.menu} onClick={openOrCloseNav}>
          <MenuIcon style={{ fontSize: 30 }} />
        </div>
        <div>
          <Link href="/">
            <a className={styles.logo}>ShopShop</a>
          </Link>
        </div>
        <div className={styles.nav}>
          <Nav open={openNav} />
          <div className={styles.cart} onClick={showCartHandler}>
            <ShoppingCartOutlinedIcon style={{ fontSize: 30 }} />
            {basket.length > 0 ? (
              <p className={styles.cartCount}>{totalBasketItems()}</p>
            ) : null}
          </div>
        </div>
      </div>
      <Cart show={showCart} close={closeCartHandler} />
      <Backdrop show={showCart} click={closeCartHandler} />
    </div>
  );
}

export default Header;
