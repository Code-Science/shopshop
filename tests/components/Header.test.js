import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Header from '../../components/Header';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('Header Component', () => {
  const wrapper = shallow(<Header />);

  it('expect to render Header component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
