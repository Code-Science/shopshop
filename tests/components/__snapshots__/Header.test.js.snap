// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot test 1`] = `
<div
  className="header"
>
  <div
    className="auth"
  >
    <button
      className="items"
      onClick={[Function]}
      type="button"
    >
      Sign In
    </button>
  </div>
  <div
    className="mainHead"
  >
    <div
      className="menu"
      onClick={[Function]}
    >
      <MenuIcon
        style={
          Object {
            "fontSize": 30,
          }
        }
      />
    </div>
    <div>
      <Link
        href="/"
      >
        <a
          className="logo"
        >
          ShopShop
        </a>
      </Link>
    </div>
    <div
      className="nav"
    >
      <Nav
        open={false}
      />
      <div
        className="cart"
        onClick={[Function]}
      >
        <ShoppingCartOutlinedIcon
          style={
            Object {
              "fontSize": 30,
            }
          }
        />
        <p
          className="cartCount"
        >
          NaN
        </p>
      </div>
    </div>
  </div>
  <Cart
    close={[Function]}
    show={false}
  />
  <Backdrop
    click={[Function]}
    show={false}
  />
</div>
`;
